src/parser.ts

```diff
@@ -1789,10 +1813,14 @@ export class Parser {
                     statement = this.parseExportDeclaration();
                     break;
                 case 'import':
-                    if (!this.context.isModule) {
-                        this.tolerateUnexpectedToken(this.lookahead, Messages.IllegalImportDeclaration);
+                    if (this.matchImportCall()) {
```

Comment:
That's my interpretation, not sure if it's correct or not. @domenic do you have an opinion?